parsers/CA_SK.py

```diff
@@ -1,142 +1,108 @@
-                "production": production_data,
-                "source": "saskpower.com",
-            }
+    date = datetime.strptime(payload["SupplyDataText"], "%b %d, %Y")
+    production_mix = ProductionMix(
+        **{
+            PRODUCTION_MAPPING[value["type"]]: value["totalGenerationForType"]
+            for value in payload["PowerCacheData"]["generationByType"]
+        }
+    )
+    production_breakdown_list = ProductionBreakdownList(logger)
+    # Convert the daily average returned by the API into hourly values. This is
+    # a bit of a hack, but it's required because the back-end requires hourly
+    # datapoints while the API only provides daily averages.
+    for hour in range(24):
+        production_breakdown_list.append(
+            datetime=date.replace(hour=hour, tzinfo=TIMEZONE),
+            production=production_mix,
+            source="saskpower.com",
+            zoneKey=ZoneKey(zone_key),
         )
-
-    return data_list
+    return production_breakdown_list.to_list()
 
 
 def fetch_consumption(
-    zone_key: str = "CA-SK",
+    zone_key: ZoneKey = ZoneKey("CA-SK"),
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-):
-    # Validate that the zone key is equal to CA-SK.
-    validate_zone_key(zone_key)
-    # Validate that the target_datetime is None as this parser is unable to fetch historical data.
-    validate_no_datetime(target_datetime, zone_key)
-
-    session = session or Session()
-
-    # Set the headers to mimic a user browser as the API will return a 403 if not.
-    headers = {"user-agent": USER_AGENT}
-
-    response: Response = session.get(CONSUMPTION_URL, headers=headers)
-
-    if not response.ok:
-        raise ParserException(
-            "CA_SK.py",
-            f"Failed to fetch consumption data. Response Code: {response.status_code}\nError:\n{response.text}",
-            zone_key,
-        )
-
-    raw_data = response.json()
-
+) -> dict[str, typing.Any]:
+    payload = _request(session, target_datetime, CONSUMPTION_URL, zone_key)
+    # The source refreshes every 5 minutes, so we assume the current data is
+    # from 5 minutes before the most recent multiple of 5 minutes.
     now = datetime.now(TIMEZONE)
-
-    # Data is updated every 5 minutes so we assume the data is from a multiple of 5 minutes and has a 5 minute delay from that multiple.
     assumed_datetime = now.replace(second=0, microsecond=0) - timedelta(
-        minutes=(now.minute % 5) + 5
+        minutes=(5 + now.minute % 5)
     )
-
-    return [
-        {
-            "zoneKey": zone_key,
-            "datetime": assumed_datetime,
-            "consumption": int(raw_data),
-            "source": "saskpower.com",
-        }
-    ]
+    return TotalConsumption.create(
+        consumption=float(payload),
+        datetime=assumed_datetime,
+        logger=logger,
+        source="saskpower.com",
+        zoneKey=zone_key,
+    ).to_dict()
```

Comment:
This should use the append function instead of the create function. The create function is an internal function that don't return anyting (it assings things internally).

Reply:
I'm not sure how that would look; when I tried it raised an `AttributeError: 'TotalConsumption' object has no attribute 'append'`. Could you give me an example of what you intend? What I have seems to work fine if I run the consumption parser with some hard coded payload, e.g., `payload = "37"`.

Reply:
The thing is that the new parser classes where designed with the assumption that all data should be lists or multiple data points (this works even for single datapoints) so it should looks something like:
```python
totalConsumption = TotalConsumption(logger)
totalConsumption.append(/* ... */)
return totalConsumption.to_list()
```

Once all parsers have been upgraded we plan on deprecating a dict as the return signature.

Reply:
I can't get your snippet to work unless I replace `TotalConsumption` with `TotalConsumptionList`, so I went with that. Let me know if that's not what you meant!